### DISCUSSION TITLE
Semaphore not firing

### DISCUSSION THREAD

**Discussion by orbitcoms** [Mair/esp32-course#103] [Course Qeustions]
Hi,
I could not find any information in the RTOS videos how to use semaphores across files. I cannot get a semaphore being Given in one file to get taken in another.

I have Wifi.C file that defines and gives the semaphore like this.

xSemaphoreHandle wifi_data;

in wifi receive function...
xSemaphoreGive(wifi_data);  //I create a print before this line to confirm code is getting here

void init()...
wifi_data = xSemaphoreCreateBinary();

My other file "Process.C" I set up like this...

#include "wifi.h"

extern xSemaphoreHandle wifi_data;

And my task...
void process_wifi_detection(void * params)
{
	while (true)
	{
		xSemaphoreTake(wifi_data, portMAX_DELAY);

**Comment by Mair** [comment]
@orbitcoms It looks good to me. It should fire the way you have it set up
